Game/MyProject/Game/SceneManagement/Mesh.hpp

```diff
@@ -38,19 +39,48 @@ class Mesh : std::enable_shared_from_this<Mesh> {
         _vertex_buffer = IvRenderer::mRenderer->GetResourceManager()->CreateVertexBuffer(vertex_format, (unsigned int)vertices.size(), (void*)vertices.data(), kImmutableUsage);
         
         IvStackAllocator::mScratchAllocator->Reset(current_offset);
+        
+        // sets the starting values
+        _min_max_vertices.resize(3);
+        _min_max_vertices[0].x = vertices[0].position.x;
+        _min_max_vertices[0].y = vertices[0].position.x;
+        _min_max_vertices[1].x = vertices[0].position.y;
+        _min_max_vertices[1].y = vertices[0].position.y;
+        _min_max_vertices[2].x = vertices[0].position.z;
+        _min_max_vertices[2].y = vertices[0].position.z;
+        
+        // calculates the min and max for each axe
+        for (T i : vertices) {
+            if (i.position.x < _min_max_vertices[0].x)
+                _min_max_vertices[0].x = i.position.x;
+            if (i.position.x > _min_max_vertices[0].y)
+                _min_max_vertices[0].y = i.position.x;
+            
+            if (i.position.y < _min_max_vertices[1].x)
+                _min_max_vertices[1].x = i.position.y;
+            if (i.position.y > _min_max_vertices[1].y)
+                _min_max_vertices[1].y = i.position.y;
+            
+            if (i.position.z < _min_max_vertices[2].x)
+                _min_max_vertices[2].x = i.position.z;
+            if (i.position.z > _min_max_vertices[2].y)
+                _min_max_vertices[2].y = i.position.z;
+        }
     }
     
     void setIndexBuffer(const std::vector<unsigned int>&);
     
     IvVertexBuffer* getVertexBuffer();
     IvIndexBuffer* getIndexBuffer();
     const unsigned int& getBufferSize() const;
+    const std::vector<IvVector2>& getMinMaxVertices() const;
     
 private:
     // private variable(s)
-    IvVertexBuffer* _vertex_buffer;
-    IvIndexBuffer*  _index_buffer;
-    unsigned int    _buffer_size;
+    IvVertexBuffer*         _vertex_buffer;
+    IvIndexBuffer*          _index_buffer;
+    std::vector<IvVector2>  _min_max_vertices;
```

Comment:
Two vectors, IvVector3 for min and max is enough. Using std::vector if inefficient for this.